portal-kernel/src/com/liferay/portal/kernel/service/persistence/impl/BasePersistenceImpl.java

```diff
@@ -699,18 +701,22 @@ public T update(T model) {
 		boolean isNew = model.isNew();
 
 		if (isNew && (_dataLimitModelMaxCount > 0)) {
-			DynamicQuery dynamicQuery = DynamicQueryFactoryUtil.forClass(
-				clazz, clazz.getClassLoader());
-
 			AuditedModel auditedModel = (AuditedModel)model;
 
-			dynamicQuery.add(
-				RestrictionsFactoryUtil.eq(
-					"companyId", auditedModel.getCompanyId()));
+			FromStep fromStep = DSLQueryFactoryUtil.count();
+
+			JoinStep joinStep = fromStep.from(_table);
+
+			GroupByStep groupByStep = joinStep.where(
+				_table.getColumn(
```

Comment:
They already had a bug here assuming there's a companyId column for every entity.  Can fix using DSL with a simple null check for the column.